openwisp_notifications/tasks.py

```diff
@@ -25,3 +31,66 @@ def delete_obsolete_notifications(instance_app_label, instance_model, instance_i
         | Q(target_object_id=instance_id)
     )
     Notification.objects.filter(where).delete()
+
+
+# Following tasks add notification settings in database
+
+
+@shared_task
+def ns_register_notification_type(notification_type):
+    """
+    Adds notification setting for all users who are memeber of an
+    organization when a new notification type is registered.
+    """
+    for user in User.objects.all():
+        if not user.organizations_dict.keys():
+            try:
+                NotificationSetting.objects.create(
+                    user_id=user.id, type=notification_type, organization_id=None
+                )
+            except IntegrityError:
+                pass
+            finally:
+                continue
+
+        for org_user in user.organizations_dict:
+            try:
+                NotificationSetting.objects.create(
+                    user_id=user.id, organization_id=org_user, type=notification_type,
+                )
+            except IntegrityError:
+                continue
+
+
+@shared_task
+def ns_unregister_notification_type(notification_type):
+    """
+    Deletes all notification settings related to a unregistered
+    notification type.
+    """
+    NotificationSetting.objects.filter(type=notification_type).delete()
+
+
+@shared_task
+def ns_organization_user_added(instance_user_id, instance_org_id):
+    """
+    Adds notification settings for all notification types when a new
+    organization user is added.
+    """
+    for notification_type in NOTIFICATION_TYPES.keys():
+        NotificationSetting.objects.create(
+            user_id=instance_user_id,
+            organization_id=instance_org_id,
+            type=notification_type,
+        )
+
+
+@shared_task
+def ns_organization_user_deleted(instance_user_id, instance_org_id):
+    """
+    Deletes notification settings for all notification types when
+    an organization user is deleted.
+    """
+    NotificationSetting.objects.filter(
+        user_id=instance_user_id, organization_id=instance_org_id
+    ).delete()
```

Comment:
are you sure this is necessary? If the `organization` field is set, the SQL DELETE should cascade.

Reply:
It is a foreign key to `Organization` not `OrganizationUser`.